### ISSUE TITLE
'check_version_conflict': using 'rails new', with a version, when Rails 5.1.X installed

### ISSUE THREAD

**Issue by matthewd** [rubygems/rubygems#1931]
I'm having a problem or would like to suggest a feature.

My current problem is described in https://github.com/rails/rails/issues/29113. In brief: with multiple versions of a gem installed, when explicitly choosing to run a command from an older one, a subsequent require can incorrectly attempt to activate a newer version of a gem after the older version is already activated.

I have reproduced the behaviour with Ruby 2.4.1 + RubyGems 2.6.12:

```
$ gem install rails -v 5.1.1
$ gem install rails -v 5.0.0.1
$ rails _5.0.0.1_ -v
/Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/specification.rb:2278:in `check_version_conflict': can't activate activesupport-5.1.1, already activated activesupport-5.0.0.1 (Gem::LoadError)
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/specification.rb:1404:in `activate'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:89:in `block in require'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:88:in `each'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:88:in `require'
	from /Users/matthew/tmp/x/gems/concurrent-ruby-1.0.5/lib/concurrent.rb:15:in `<top (required)>'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/tmp/x/gems/activesupport-5.0.0.1/lib/active_support/logger_silence.rb:3:in `<top (required)>'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/tmp/x/gems/activesupport-5.0.0.1/lib/active_support/logger.rb:1:in `<top (required)>'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/tmp/x/gems/activesupport-5.0.0.1/lib/active_support.rb:27:in `<top (required)>'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/tmp/x/gems/railties-5.0.0.1/lib/rails/generators.rb:6:in `<top (required)>'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/tmp/x/gems/railties-5.0.0.1/lib/rails/commands/application.rb:1:in `<top (required)>'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/tmp/x/gems/railties-5.0.0.1/lib/rails/cli.rb:14:in `<top (required)>'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/.rbenv/versions/2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/matthew/tmp/x/gems/railties-5.0.0.1/exe/rails:9:in `<top (required)>'
	from /Users/matthew/tmp/x/bin/rails:23:in `load'
	from /Users/matthew/tmp/x/bin/rails:23:in `<main>'
```

I described a theory in https://github.com/rails/rails/issues/29113#issuecomment-303599678, but that seems flawed, because it offers no explanation for why this has just arisen recently.

**Comment by segiddins** [comment]
Can you please put a full description of the problem, including steps to reproduce, into the issue body? Thanks!

**Comment by tenderlove** [comment]
Seems like this is fixed on master with e015540b46d3b614b6ad5a5363b2f1395d01bf74

**Comment by segiddins** [comment]
That PR shipped in 2.6.12

**Comment by kurko** [comment]
Seems to be fixed in 2.6.12

```bash
rails _5.0.2_ --version
/Users/kurko/.rbenv/versions/2.3.1/lib/ruby/site_ruby/2.3.0/rubygems/specification.rb:2278:in `check_version_conflict': can't activate activesupport-5.1.1, already activated activesupport-5.0.2 (Gem::LoadError)
```

**Comment by hsbt** [comment]
This is already fixed.
